hamilton/function_modifiers/adapters.py

```diff
@@ -0,0 +1,543 @@
+        dependencies, resolved_kwargs = resolve_kwargs(self.kwargs)
+        dependencies_inverted = {v: k for k, v in dependencies.items()}
+
+        def save_data(
+            __adapter_factory=adapter_factory,
+            __dependencies=dependencies_inverted,
+            __resolved_kwargs=resolved_kwargs,
+            __data_node_name=node_.name,
+            **input_kwargs,
+        ) -> Dict[str, Any]:
+            input_args_with_fixed_dependencies = {
+                __dependencies.get(key, key): value for key, value in input_kwargs.items()
+            }
+            kwargs = {**__resolved_kwargs, **input_args_with_fixed_dependencies}
+            data_to_save = kwargs[__data_node_name]
+            kwargs = {k: v for k, v in kwargs.items() if k != __data_node_name}
+            data_saver = __adapter_factory.create_saver(**kwargs)
+            return data_saver.save_data(data_to_save)
+
+        def get_input_type_key(key: str) -> str:
+            return key if key not in dependencies else dependencies[key]
+
+        input_types = {
+            get_input_type_key(key): (Any, DependencyType.REQUIRED)
+            for key in saver_cls.get_required_arguments()
+        }
+        input_types.update(
+            {
+                (get_input_type_key(key) if key not in dependencies else dependencies[key]): (
+                    Any,
+                    DependencyType.OPTIONAL,
+                )
+                for key in saver_cls.get_optional_arguments()
+            }
+        )
+        # Take out all the resolved kwargs, as they are not dependencies, and will be filled out
+        # later
+        input_types = {
+            key: value for key, value in input_types.items() if key not in resolved_kwargs
+        }
+        input_types[node_.name] = (node_.type, DependencyType.REQUIRED)
+
+        save_node = node.Node(
+            name=artifact_name,
+            callabl=save_data,
+            typ=Dict[str, Any],
+            input_types=input_types,
+            namespace=artifact_namespace,
+        )
+        return [save_node, node_]
+
+    def validate(self, fn: Callable):
+        pass
+
+
+class save_to(metaclass=save_to__meta__):
+    """Decorator that outputs data to some external source. You can think
+    about this as the inverse of load_from.
+
+    This decorates a function, takes the final node produced by that function
+    and then appends an additional node that saves the output of that function.
+
+    As the load_from decorator does, this decorator can be referred to in a
+    dynamic way. For instance, @save_to.json will save the output of the function
+    to a json file. Note that this means that the output of the function must
+    be a dictionary (or subclass thereof), otherwise the decorator will fail.
+
+    Looking at the json example:
+
+    .. code-block:: python
+
+        @save_to.json(path=source("raw_data_path"), artifact_="data_save_output")
```

Comment:
yeah I don't like `artifact_` here.

What is bothering me is:

1. it's the name to request it as output from the driver -- and nothing to do with the actual artifact name itself, e.g. the file name. So `artifact_` is a little misleading?
2. `name_` could also be just as ambiguous? 
3. `output_name_` could be better?
4. `hamilton_name_` ?

Reply:
Agreed, waiting on response in the hamilton community. `access_name_` maybe? `refer_to_`?

Reply:
Chose `node_name_` for now as its pretty clear what's happening... Can change before release.